DietPi-Software | Restic: is a fast and secure backup program

goldfix opened the pull request:
Added new backup software:
- Restic: is a fast and secure backup program

goldfix committed:
added script to install and remove Restic

goldfix committed:
added call to force autoupdate

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4568,6 +4574,18 @@ The install script will now exit. After applying one of the the above, rerun die
+			# APT Update
+			G_AGUP
+
+			# APT package
```
```suggestion
			# APT package
```
This is not needed as it is done once before all software installs start.

goldfix committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng committed:
Merge branch 'dev' into pr/6350

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -4594,6 +4600,15 @@ The install script will now exit. After applying one of the the above, rerun die
+			G_AGI restic
+
+			# Force autoupdate
+			G_EXEC restic self-update
```
This requires a change in the `Download_Install` function, I hope to find time to implement it later:
```suggestion
			case $G_HW_ARCH in
				3) local arch='arm64';;
				10) local arch='amd64';;
				*) local arch='arm';;
			esac

			local fallback_url="https://github.com/restic/restic/releases/download/v0.15.2/restic_0.15.2_linux_$arch.bz2"
			Download_Install "$(curl -sSfL 'https://api.github.com/repos/restic/restic/releases/latest' | mawk -F\" "/^ * \"browser_download_url\": \".*\/restic_[^\"\/]*_linux_$arch\.bz2\"$/{print \$4}")" /usr/local/bin/restic
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -14069,6 +14084,11 @@ _EOF_
 
+		if To_Uninstall 207 # Restic
+		then
+			G_AGP restic
```
```suggestion
			[[ -f '/usr/local/bin/restic' ]] G_EXEC rm /usr/local/bin/restic
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -759,6 +759,12 @@ Available commands:
 		#--------------------------------------------------------------------------------
+		software_id=207
+		aSOFTWARE_NAME[$software_id]='Restic'
+		aSOFTWARE_DESC[$software_id]='Restic is a modern backup program that can back up your files'
```
Following closer the description on their GitHub + making clear that it is CLI: https://github.com/restic/restic
```suggestion
		aSOFTWARE_DESC[$software_id]='Fast, efficient and secure command-line backup program'
```

MichaIng committed:
Merge branch 'dev' into pr/6350

MichaIng committed:
v8.17

- DietPi-Software | Restic: Follow closer the description on their GitHub + making clear that it is a CLI tool: https://github.com/restic/restic

MichaIng committed:
v8.17

- DietPi-Software | Restic: Download binary from upstream repo in the first place and skip APT install

MichaIng committed:
v8.17

- DietPi-Software | Restic: Update uninstall code for new install method

MichaIng committed:
v8.17

- DietPi-Software | Restic: Add meta info, changelog, re-arrange code to be located behind Rclone, add RISC-V support

MichaIng committed:
v8.17

- DietPi-Software | Restic: Fix install ID

goldfix committed:
Merge branch 'dev' into dev-add-restic-software
